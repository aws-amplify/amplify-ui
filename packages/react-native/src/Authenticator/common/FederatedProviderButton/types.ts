import {
  ImageSourcePropType,
  ImageStyle,
  TextStyle,
  ViewStyle,
} from 'react-native';

import { ButtonProps } from '../../../primitives';

export interface FederatedProviderButtonProps extends ButtonProps {
  source: ImageSourcePropType;
}

export interface FederatedProviderButtonStyles {
  container: ViewStyle;
  icon: ImageStyle;
  label: TextStyle;
}
