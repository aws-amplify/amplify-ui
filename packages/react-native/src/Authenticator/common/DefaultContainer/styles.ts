import { StyleSheet, ViewStyle } from 'react-native';
import { StrictTheme } from '../../../theme';

import { ContainerStyles, InnerContainerStyles } from './types';

export const getThemedStyles = (
  { tokens: { colors } }: StrictTheme,
  insetPadding: Pick<
    ViewStyle,
    'paddingBottom' | 'paddingLeft' | 'paddingRight' | 'paddingTop'
  >
): Required<ContainerStyles> => {
  return StyleSheet.create({
    keyboardAvoidingView: {
      alignContent: 'center',
      backgroundColor: colors.background.primary,
      justifyContent: 'center',
      flex: 1,
      // TDOO: add theme override
    },
    scrollViewContentContainer: {
      ...insetPadding,
      alignContent: 'center',
      justifyContent: 'center',
      flexGrow: 1,
      // TDOO: add theme override
    },
    scrollView: {}, // TDOO: add theme override
  });
};

export const getInnerContainerStyles = ({
  tokens: { space },
}: StrictTheme): InnerContainerStyles =>
  StyleSheet.create({
    container: {
      marginHorizontal: space.medium,
      // TDOO: add theme override
    },
  });
