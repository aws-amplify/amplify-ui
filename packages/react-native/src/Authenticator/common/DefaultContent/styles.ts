import { StyleSheet } from 'react-native';

import { StrictTheme } from '../../../theme';
import { DefaultContentStyle } from './types';

export const getDefaultStyle = ({
  tokens: { colors, fontSizes, space },
}: StrictTheme): Required<DefaultContentStyle> =>
  StyleSheet.create({
    body: {
      fontSize: fontSizes.medium,
      paddingHorizontal: space.small,
    },
    buttonPrimary: {
      margin: space.small,
    },
    buttonPrimaryLabel: {}, // themed value only
    buttonSecondary: {
      marginHorizontal: space.medium,
    },
    buttonSecondaryLabel: {}, // themed value only
    errorMessage: {
      marginHorizontal: space.small,
    },
    errorMessageIcon: {}, // themed value only
    errorMessageLabel: {}, // themed value only
    fieldContainer: {
      paddingHorizontal: space.small,
    }, // themed value only
    fieldErrorsContainer: {
      paddingHorizontal: space.medium,
      paddingVertical: space.xs,
    },
    fieldError: {
      color: colors.font.error,
      paddingVertical: 1,
    }, // themed value only
    fieldLabel: {}, // themed value only
    fieldStyle: {}, // themed value only
    footer: {}, // themed value only
    formFields: {
      paddingBottom: space.xs,
    },
    header: {
      marginVertical: space.small,
      paddingHorizontal: space.small,
    },
    link: {
      marginVertical: space.xs,
      minWidth: '50%',
    },
    linksContainer: {
      flexDirection: 'row',
      flexWrap: 'wrap',
      justifyContent: 'center',
    },
  });
