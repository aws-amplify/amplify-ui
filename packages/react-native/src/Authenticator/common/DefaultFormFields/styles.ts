import { StyleSheet } from 'react-native';

import { DefaultFormFieldStyle } from './types';

export const styles: DefaultFormFieldStyle = StyleSheet.create({
  field: {
    fontSize: 16,
    marginVertical: 4,
  },
  error: {
    color: 'red',
  },
});
