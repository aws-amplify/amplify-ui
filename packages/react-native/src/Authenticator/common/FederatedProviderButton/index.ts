export { default as FederatedProviderButton } from './FederatedProviderButton';
export { FederatedProviderButtonProps } from './types';
