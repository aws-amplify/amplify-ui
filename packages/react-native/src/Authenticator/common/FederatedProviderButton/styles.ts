import { StyleSheet } from 'react-native';

import { FederatedProviderButtonStyles } from './types';

export const styles: FederatedProviderButtonStyles = StyleSheet.create({
  container: {
    alignItems: 'center',
    borderRadius: 4,
    borderWidth: StyleSheet.hairlineWidth,
    flexDirection: 'row',
    justifyContent: 'center',
    paddingVertical: 8,
    width: '100%',
  },
  icon: {
    paddingHorizontal: 16,
  },
  label: {
    fontWeight: '400',
  },
});
