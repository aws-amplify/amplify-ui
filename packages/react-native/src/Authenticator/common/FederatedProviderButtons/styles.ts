import { FederatedProviderButtonStyle } from './types';

export const styles: FederatedProviderButtonStyle = {
  button: { marginVertical: 8 },
  container: { paddingVertical: 4 },
  text: { paddingVertical: 8, textAlign: 'center' },
};
