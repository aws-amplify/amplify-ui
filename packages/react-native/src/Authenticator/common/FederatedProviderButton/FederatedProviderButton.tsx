import React from 'react';

import { Button, Icon, Label } from '../../../primitives';
import { usePressableContainerStyles } from '../../../hooks';

import { styles } from './styles';
import { FederatedProviderButtonProps } from './types';

export default function FederatedProviderButton({
  children,
  source,
  style,
  textStyle,
  ...rest
}: FederatedProviderButtonProps): JSX.Element {
  const pressableStyle = usePressableContainerStyles({
    containerStyle: styles.container,
    overrideStyle: style,
  });

  return (
    <Button {...rest} style={pressableStyle}>
      <Icon source={source} style={styles.icon} size={20} />
      <Label style={[styles.label, textStyle]}>{children}</Label>
    </Button>
  );
}
