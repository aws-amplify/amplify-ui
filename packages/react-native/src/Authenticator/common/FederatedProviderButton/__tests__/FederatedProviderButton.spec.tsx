import React from 'react';
import { fireEvent, render, renderHook } from '@testing-library/react-native';

import { icons } from '../../../../assets';
import FederatedProviderButton from '../FederatedProviderButton';
import { useTheme } from '../../../../theme';
import { getThemedStyles as getButtonThemedStyles } from '../../../../primitives/Button/styles';
import { styles } from '../styles';

const signInText = 'Sign In with Amazon';

describe('FederatedProviderButton', () => {
  it('renders default button as expected', () => {
    const { toJSON, getByRole } = render(
      <FederatedProviderButton source={icons.amazonLogo}>
        {signInText}
      </FederatedProviderButton>
    );
    expect(toJSON()).toMatchSnapshot();

    expect(getByRole('button')).toBeDefined();
    const icon = getByRole('image');
    expect(icon).toBeDefined();
    expect(icon.props.source).toEqual(icons.amazonLogo);
  });

  it('handles an onPress callback', () => {
    const toFederatedSignInMock = jest.fn();
    const { getByText } = render(
      <FederatedProviderButton
        source={icons.amazonLogo}
        onPress={toFederatedSignInMock}
      >
        {signInText}
      </FederatedProviderButton>
    );

    const button = getByText(signInText);
    fireEvent.press(button);

    expect(toFederatedSignInMock).toHaveBeenCalledTimes(1);
  });

  it('applies style props', () => {
    const customStyle = { backgroundColor: 'blue' };

    const { toJSON, getByRole } = render(
      <FederatedProviderButton source={icons.amazonLogo} style={customStyle}>
        {signInText}
      </FederatedProviderButton>
    );

    const { result } = renderHook(() => useTheme());
    const buttonThemedStyle = getButtonThemedStyles(result.current);
    expect(getByRole('button').props.style).toStrictEqual([
      { ...buttonThemedStyle.container, ...buttonThemedStyle.containerDefault },
      undefined, // button pressed styles
      [
        styles.container,
        undefined, // pressed style
        customStyle,
      ],
    ]);

    expect(toJSON()).toMatchSnapshot();
  });
});
