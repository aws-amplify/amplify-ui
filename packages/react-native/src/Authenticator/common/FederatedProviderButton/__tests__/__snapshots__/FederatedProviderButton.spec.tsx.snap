// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FederatedProviderButton applies style props 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "hsl(210, 10%, 58%)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "gap": 12,
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      },
      undefined,
      [
        undefined,
        undefined,
        {
          "backgroundColor": "blue",
        },
      ],
    ]
  }
>
  <Image
    accessibilityRole="image"
    accessible={true}
    source={
      {
        "uri": "amazonLogo",
      }
    }
    style={
      [
        {
          "height": 20,
          "resizeMode": "contain",
          "tintColor": undefined,
          "width": 20,
        },
        undefined,
      ]
    }
  />
  <Text
    accessibilityRole="text"
    style={
      [
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        {
          "color": "hsl(210, 50%, 10%)",
        },
        undefined,
      ]
    }
  >
    Sign In with Amazon
  </Text>
</View>
`;

exports[`FederatedProviderButton renders default button as expected 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "hsl(210, 10%, 58%)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "gap": 12,
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      },
      undefined,
      [
        undefined,
        undefined,
        undefined,
      ],
    ]
  }
>
  <Image
    accessibilityRole="image"
    accessible={true}
    source={
      {
        "uri": "amazonLogo",
      }
    }
    style={
      [
        {
          "height": 20,
          "resizeMode": "contain",
          "tintColor": undefined,
          "width": 20,
        },
        undefined,
      ]
    }
  />
  <Text
    accessibilityRole="text"
    style={
      [
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        {
          "color": "hsl(210, 50%, 10%)",
        },
        undefined,
      ]
    }
  >
    Sign In with Amazon
  </Text>
</View>
`;
